package.json

```diff
@@ -42,6 +42,9 @@
     "node": ">=0.10.3"
   },
   "scripts": {
+    "commitmsg": "commitlint -e $GIT_PARAMS",
+    "precommit": "bash scripts/pre-commit.sh",
+    "prepush": "bnr lint && bnr test && bnr build && bnr docs",
```

Comment:
imho I think prepush is not necessary. I dont want to wait to be able to push to remote branches. We should create pr and let ci do the work. Another option is to make this optional and disabled by default and document it in readme. Let ppl decide if they wan to use it or not.

Reply:
Looks like there's some disagreement within the team about which hooks to use:

> I hate waiting when I commit (https://github.com/char0n/ramda-adjunct/issues/386#issuecomment-366721586, @guillaumearm)

Also, I think that this hook could be optimized anyway:

> `npm run build` creates 3 builds. Trying to build just one just to see if it succeeds would probably suffice on pre-push, especially when all of the builds are created on `prepublishOnly`.

And, I'm not sure if trying to build is needed anyway, `npm test` works somehow even after `npm run clean`. And I can't tell what `bnr docs` actually does.

And, it has an issue:

> It probably runs everything on version from working dir instead of from last commit -- someone could fix the issues and forget to commit them, and push wouldn't stop him. (https://github.com/char0n/ramda-adjunct/issues/381#issuecomment-367982987)

> Another option is to make this optional [...]. Let ppl decide if they wan to use it or not.

Interesting idea. My take on this, we could make the hooks enabled by default and allow to disable each one of them separately by setting an env variable.

Reply:
The fact that there is a CI set up is a valid point.

Reply:
@char0n : prepush hook is an habit from me, when I have no CI on my project, so it's fine, we can drop it.

@GingerPlusPlus : I'm ready to wait for `commitlint` and `eslint`, don't worry about me. next step is #313.